### DISCUSSION TITLE
Is the ESP32-C3F ADC supported yet?

### DISCUSSION THREAD

**Discussion by brev-dev** [arendst/Tasmota#13155] [Q&A] [answered]
I'm experimenting with Ai-Thinker ESP32-C3F's, with the objective of using them as a relatively easy swap-in on my [IOT sensor board](https://github.com/brev-dev/another_esp8266_sensor_board) (more available ports for things like multiple moisture sensors, etc.). They're currently just plugged into esp8266 flasher boards while testing. 

I'm using the latest devel builds, with the help of @Jason2866's [web installer](https://jason2866.github.io/Tasmota-specials/).

Buttons/relays/switches are working fine, as is the internal temperature sensor after a recent update. But I've had no luck getting sensible readings from the ADC. Analog readings are always around 4000, whatever I do.

The c3 stuff is so new, not sure what the best forum is for this question? Here? Discord? Raising an issue?

Thanks!

**Comment by Jason2866** [comment, marked as answer]
@brev-dev Discussion is correct. Since the Tasmota C3 variant is still experimental, we would close a issue.
@s-hadinger can you have a look? Scanning through Arduino ESP32 github issues, it could! work already

**Reply by s-hadinger** [reply]
I haven't tried ADC on C3 yet. I will have a look.

@brev-dev which GPIO are you using for ADC?

**Reply by brev-dev** [reply]
Thanks, @s-hadinger.
I've been testing every combination I can, but they all seem to (not) respond in the same way. Given I'm using the c3f, this means I've tried:

| Pin | Port | Comments |
| --- | ---|---|
|1|GPIO0/ADC1_0|This is reset on esp8266; haven't checked if there's some flasher-board wiring that might interfere with the ADC on this one|
|2|GPIO1/ADC1_1|This is the esp8266's ADC pin#
|4|GPIO2/ADC1_2/SPI_Q(MISO)|
|5|GPIO3/ADC1_3|
|6|GPIO4/ADC1_4/SPI_HD|
|7|GPIO5/ADC2_0/SPI_WP|I think ADC2 is shared with WiFi, so maybe I wouldn't expect this one to work?|

In Tasmota, you then get to choose a number for the Analog in the pulldown. Don't know if this has any significance other than a label to assign to the measurement? Anyway, I've tried changing this number, again with no apparent effect.

**Reply by s-hadinger** [reply]
Thanks, indeed ADC2 is not usable when wifi is on

**Reply by brev-dev** [reply]
Any luck with this? As of build date "2021-09-22T15:35:37" I'm still seeing a while lot of nothing (see below). It'd be great to know if someone else is seeing the same, to confirm that it's not just a mistake on my end, or a problem with my batch of c3f's.

![image](https://user-images.githubusercontent.com/77922126/134397549-fff74f04-a000-460e-92e6-cf045fbe4170.png)

**Reply by Jason2866** [reply]
Probably your C3 are all okay. Since the Arduino Core support for the ESP32-C3 is very young there are open issues. This issue hits the S2 and the C3 https://github.com/espressif/arduino-esp32/issues/5691
Tasmota uses the Arduino core. As long this issue is not fixed Tasmota can not work correctly

**Reply by brev-dev** [reply]
Thanks!

**Reply by s-hadinger** [reply]
I took a look at Arduino and it seems that ADC support for ESP32C3 is completely missing. I will try to make it work.

**Comment by Jason2866** [comment]
Try again. I get readings between 0 and 1 when nothing is connected

**Reply by brev-dev** [reply]
Upgrading today to your C3 4mb image based on arendst/Tasmota@8605ad3, I still see the same type of numbers as the screenshot above. No obvious changes to the numbers when switching the ADC pin connections to GND vs 3.3V.

**Reply by s-hadinger** [reply]
I'm not sure Arduino did the necessary changes. We can't do anything before Arduino supports ADC on C3.

**Reply by EdBrisebois** [reply]
I have the same issue.  No matter what pin I assign for ADC, only AO works in Arduino IDE or TASMOTA. I asked "Arduino" but they do not support 3rd party chipsets/boards.  So whomever is building the ESP32-C3 support that is currently available on the Arduino IDE, has not fixed the ADC issue.  Which is a shame since the AI Thinker C3-12F is basically a drop in for the 8266-12F. I have decide to drop C3 for now.... All those 8266 boards out there in limbo :o(

**Reply by barbudor** [reply]
@EdBrisebois the ESP32 Arduino core is developed by Espressif
You can open issues here https://github.com/espressif/arduino-esp32

There are quite some issues already opened for ADC on C3

**Reply by EdBrisebois** [reply]
I know. 
If you are versed with the ESP-IDF, their tools will get the ADC going.
Of course both Espressif and AI Thinker recommend it.
I not a coder, I use arduino-IDE for it's simplicity and open user support.
Just does not have the support yet for the RISC-V chips.

**Reply by Jason2866** [reply]
It is still broken in Arduino core, you can use only `GPIO 0`  if you define this as ADC.

**Reply by Jason2866** [reply]
ADC (channel 0-4) for C3 should work now with the latest development core. Can be tested by copying `platformio_override_sample.ini` to `platformio_override.ini` and uncomment lines 93 to 96.

**Reply by brev-dev** [reply]
Yes, the current version through the web installer now gives me 5 ADC inputs, thanks!
